pallets/marketplace/src/lib.rs

```diff
@@ -466,6 +563,14 @@ pub mod pallet {
                             JobStatus::Assigned,
                         );
                     }
+                    let total_jobs_assigned =
+                        <StoredTotalJobsAssigned<T>>::get(&reward_asset).unwrap_or_default();
+                    let new_total_jobs_assigned = total_jobs_assigned + T::AssetAmount::from(1);
+
+                    <StoredTotalJobsAssigned<T>>::insert(
+                        reward_asset.clone(),
+                        new_total_jobs_assigned,
+                    );
```

Comment:
Nit-picky: you could have it in one operation like

<StoredTotalJobsAssigned<T>>::mutate(&reward_asset, |total_jobs_assigned| {
  *total_jobs_assigned = total_jobs_assigned + 1u128
});